### PR TITLE
Upload / connect split

### PR DESCRIPTION
**Pull request by alejandraarri**
In this PR we split the _Connect Dataset_ tab that was in _Add dataset/Layer_ modal window into _Upload_ and _Connect_ tabs.
## Related issue
https://github.com/CartoDB/cartodb/issues/15207

## Screenshots
Screenshot 2019-11-15 at 17 40 55

### General use case
- [ ] Access the modal from _New dataset_ button in _Dashboard_ 
   - [ ] Upload a valid file. A new dataset should be created. No new map.
   - [ ] Test URL upload. A new dataset should be created. No new map. 
(http://data.insideairbnb.com/spain/catalonia/barcelona/2019-09-17/data/listings.csv.gz)
   - [ ] CTA in _Upload_ tab should be `Upload dataset`
   - [ ] CTA in _Connect_ tab should be `Connect dataset`
 - [ ] Access the modal from _New map_ button in _Dashboard_ 
   - [ ] Upload a valid file. A new dataset and a new map should be created.
   - [ ] Test URL upload. A new dataset and a new map should be created. (http://data.insideairbnb.com/spain/catalonia/barcelona/2019-09-17/data/listings.csv.gz)
   - [ ] CTA in _Upload_ tab should be `Upload dataset`
   - [ ] CTA in _Connect_ tab should be `Connect dataset`
- [ ] Access the modal from _Add new layer_ button in a Builder map 
  - [ ] Upload a valid file. A new layer (new dataset) in our current map should be created. The dataaset should be in _Datasets_ list
   - [ ] Test URL upload.  A new layer (new dataset) in our current map should be created. The dataaset should be in _Datasets_ list (http://data.insideairbnb.com/spain/catalonia/barcelona/2019-09-17/data/listings.csv.gz)
   - [ ] CTA in _Upload_ tab should be `Add layer`
   - [ ] CTA in _Connect_ tab should be `Add layer`

### No dataset quota use case
- [ ] Change user’s limits to current dataset usage.
- [ ] Access the modal from _New dataset_ button in _Dashboard_ 
   - [ ] _Upload_ and _Import_ tabs should be disabled.
 - [ ] Access the modal from _New map_ button in _Dashboard_ 
   - [ ] _Upload_ and _Import_ tabs should be disabled.
- [ ] Access the modal from _Add new layer_ button in a Builder map
  - [ ] _Upload_ and _Import_ tabs should be disabled.

### CODE REVIEW

**Comment by alejandraarri** [comment]
Retest this please

File: lib/assets/javascripts/dashboard/views/dashboard/dialogs/create-dialog/footer/dialog-footer.tpl [hunk 1]
```diff
@@ -40,7 +40,7 @@
           <div class="js-footer-info CreateDialog-footerInfo"></div>
           <div class="CreateDialog-footerActions js-footerActions">
             <button class="CDB-Button CDB-Button--primary CreateDialog-footerActionsButton <% if (!isUploadValid) { %>is-disabled<% } %> js-connect">
-              <span class="CDB-Button-Text CDB-Text is-semibold CDB-Size-small u-upperCase">connect dataset</span>
+              <span class="CDB-Button-Text CDB-Text is-semibold CDB-Size-small u-upperCase"><%- listingState === "upload" ? 'Upload' : 'Connect' %> dataset</span>
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/dashboard/views/dashboard/dialogs/create-dialog/footer/dialog-footer.tpl]: Ok, moving them in next PR

File: lib/assets/javascripts/builder/components/modals/add-layer/content/imports/import-options.js [hunk 2]
```diff
@@ -23,17 +23,17 @@ var ImportBigQueryView = require('./import-bigquery/import-bigquery-view');
...
-    enabled: function (config, userModel) { return true; },
-    name: 'file',
-    title: 'Data file',
-    options: {
-      type: 'url',
-      fileEnabled: true,
-      acceptSync: true
-    }
-  },
+  // File: {
```
[the remark this answers is not in the document]
Comment (alejandraarri) [reply lib/assets/javascripts/builder/components/modals/add-layer/content/imports/import-options.js]: No, there are other Importers commented. So I thought to leave it commented as the others.

**Comment by csubira** [comment]
Acceptance done following the steps mentioned in the description -> 🆗 👌

**Pull request merged**